### PR TITLE
rdm: add glossary

### PR DESCRIPTION
**Pull request by haukepetersen**
in the style of #6191, I think it would be nice to add a glossary to RIOT. This could help beginners to understand our RIOT gibberish better, but we can also use this to put some expressions in the right context as we are using them, also as target for target for referencing it in the future.

The current state is very early, wanted to get a general opinion about such a document first.

### CODE REVIEW

File: doc/memos/rdm1.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+***
+RIOT Developer Memo Format, Publishing and Maintenance Process
+***
+
+# Abstract
+
+This memo contains a glossary with expressions commonly used in RIOT and in the
+context of RIOT. The aim of this document is not only to put some names to often
+used abbreviations, but also to explain their interpretation withing the RIOT
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: s/withing/within/

File: doc/memos/rdm1.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+# Terminology
+
+See below.
+
+
+# 1. Introduction
+
+As with many open source communities, there are a number of expressions, that
+are being commonly used among community members and community documents. Though
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: s/are being/are/

File: doc/memos/rdm1.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+most of these expressions are well known among developers, some of them are
+specific or are used with not-so-common semantic in the context of RIOT.
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: remove dashes

File: doc/memos/rdm1.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+This document provides a list of the most commonly used abbreviations and
+bash-words in RIOT. The scope of this document is not only to map these
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: Does the term "bash-word" exist?

File: doc/memos/rdm1.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+**ACK:** Acknowledgment of a pull request (PR) from a RIOT maintainer.
+
+**IoT:** Internet of Things - the thing everyone talks about theses days. In the
+context of RIOT we consider the IoT to be about a huge number of constrained,
+interconnected devices, many operated on small power budgets (e.g. battery
+driven).
+
+**PR:** Git merge request on github. If you want to get code into RIOT's master
+branch, opening a PR is the only way.
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: I would add a link to https://help.github.com/articles/about-pull-requests/

File: doc/memos/rdm1.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+Another aim of this RDM is further to act as a referencing target for future any
+RMDs.
+
+
+# 2. RIOT Glossary
+
+**ACK:** Acknowledgment of a pull request (PR) from a RIOT maintainer.
+
+**IoT:** Internet of Things - the thing everyone talks about theses days. In the
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: I would rephrase. I guess these memos are supposed to survive the current hype. How about something like:
"The Internet of things has many different definitions dependent on the context. In the context of RIOT..."

File: doc/memos/rdm1.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+context of RIOT we consider the IoT to be about a huge number of constrained,
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: We should add the *heterogeneity* aspect here and directly name the most pressing constraints, i.e. add memory and CPU constraints.

File: doc/memos/rdm1.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+interconnected devices, many operated on small power budgets (e.g. battery
+driven).
+
+**PR:** Git merge request on github. If you want to get code into RIOT's master
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: I think "GitHub" is the correct spelling.

File: doc/memos/rdm1.md (hunk 5 above)
Comment (OlegHahm) [inline doc/memos/rdm1.md]: Maybe also link our development procedures.

**Comment by waehlisch** [comment]
I had a very brief look and I don't like it. Too  generic. For example, why should we explain BLE? We should expect that users know what BLE is. Users might be interested which features of BLE are supported in RIOT, or which application scenarios are enabled by BLE. Questions that should not be answered in this doc.

File: doc/memos/rdm1.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+branch, opening a PR is the only way.
+
+**RDM:** RIOT Developer Memo - these are documents like this one, containing
+typically more high-level documentation in addition to doxygen and the wiki. See
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: Not sure, if "high-level" is mandatory for these documents. Maybe "documentation of interfaces, concepts, and processes".

I would also replace "doxygen" by "API documentation" (doxygen is just the tool/format). You could also link to the API documentation and the wiki.

File: doc/memos/rdm1.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+[RDM0](TODO)
+
+**RIOT:** The name of the operating system this is all about. NOTE: it is `RIOT`,
+not `RIoT`, not `Riot`, not `riot`, and certainly not `rIoT`.
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: ...and, even more important, not "RIOT OS".

File: doc/memos/rdm1.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+
+**RIOT-MC:** RIOT Maintainers Community
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: For consistency: add a period at the end.

File: doc/memos/rdm1.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+and names of commits. Before merging code into RIOT's master, developers are
+often asked to squash their commits before a PR is merged, meaning that all the
+changes should be put in a small number of commits with clear names and touching
+one logical aspect each.
+
+TODO:
+- Commit
+- git
+- merge
+- rebase
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: I'm not sure we should explain all the git features here. IMO this is out of scope for the document.

File: doc/memos/rdm1.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+- Device
+- Flash(ing)
+- Debugg(ing)
+- JTAG
+- SWD
+- OpenOCD
+- pyterm
+- Network stack
+- 6LoWPAN
+- CoAP
```
Comment (OlegHahm) [inline doc/memos/rdm1.md]: I would also consider protocol descriptions out of scope.

**Comment by haukepetersen** [comment]
@waehlisch I agree, that many things in the proposed list are out of scope. As first draft, I simply included everything that came spontaneously to my mind. But do you oppose such a document in general, or is it just the collection of expressions you are not happy with?

**Comment by waehlisch** [comment]
@haukepetersen assuming that we keep the semantic of common terms, there should be very few RIOT-specific terms. And these are better documented on the webpage (e.g., how to spell RIOT) or in specific docs.

File: doc/memos/rdm1.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
+RIOT Maintainers Community (RIOT-MC)
+Developer Memo: 1
+Author(s): Hauke Petersen, more here
+Date: December 2016
+
+***
+RIOT Developer Memo Format, Publishing and Maintenance Process
```
Comment (miri64) [inline doc/memos/rdm1.md]: Copy-pasta from rdm0?

**Comment by haukepetersen** [comment]
I close this as I don't have a strong opinion to have this kind of document or not.

Feel free to copy the file and re-open a new PR if anyone feels like pursuing this.
